### DISCUSSION TITLE
PRs running into 403 on comment creation

### DISCUSSION THREAD

**Discussion by ascheman** [community/community#141358] [Actions]
### Select Topic Area

Question

### Body

I have some PRs from new contributors. Both run into a problem to create new comments with the auto-created GITHUB_TOKEN though we explicitely request `write` permissions.

The problem occurs [here](https://github.com/aim42/htmlSanityCheck/actions/runs/11315882998/job/31467791725) for instance with an error message like this:

```
System.Collections.Hashtable

https://api.github.com/repos/aim42/htmlSanityCheck/check-runs

System.Collections.Hashtable

Invoke-WebRequest: /home/runner/work/_actions/PavanMudigonda/jacoco-reporter/v5.0/action.ps1:338
Line |
 338 |      Invoke-WebRequest -Headers $hdr $url -Method Post -Body ($bdy | C …
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     |  {   "message": "Resource not accessible by integration",  
     | "documentation_url":
     | "https://docs.github.com/rest/checks/runs#create-a-check-run",  
     | "status": "403" }
Error: The process '/usr/bin/pwsh' failed with exit code 1
```

I had this error with [this GH action](https://github.com/aim42/htmlSanityCheck/blob/8a73b7dd27fba91dea991e646db948254bf7cfcc/.github/workflows/gradle-build.yml#L39).
A similar error occured very close in the same workflow with [that GH action](https://github.com/aim42/htmlSanityCheck/blob/8a73b7dd27fba91dea991e646db948254bf7cfcc/.github/workflows/gradle-build.yml#L32).

I could resolve both meanwhile for PRs issued by myself by [adding additional permissions](https://github.com/aim42/htmlSanityCheck/blob/8a73b7dd27fba91dea991e646db948254bf7cfcc/.github/workflows/gradle-build.yml#L1C1-L4C1):

```
permissions:
  checks: write
  pull-requests: write
```

This was recommended by both the action authors:

- in [jacoco-reporter](https://github.com/PavanMudigonda/jacoco-reporter#important-notes-) by @PavanMudigondaTR, and
- in [publish-unit-test-result](https://github.com/EnricoMi/publish-unit-test-result-action?tab=readme-ov-file#permissions) by @EnricoMi

While the latter works with additional permissions, the former still has the problem.

Any ideas appreciated.

**Comment by ascheman** [comment]
Meanwhile I got a response from GH support regarding this problem (Kudos to Arthur):

> From the example you shared, I see that the PR was opened from a fork repository. With PRs opened by forks, workflow runs triggered via the pull_request event are only able to leverage a read-only GITHUB_TOKEN for [security reasons](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#accessing-secrets).
> Workflows triggered from a forked repository using the pull_request event have read-only permissions and have no access to secrets.
> 
> 
> You can see in [this section of the job logs](https://github.com/aim42/htmlSanityCheck/actions/runs/11315882998/job/31467791725#step:1:20) that the permissions are read, which explains why a request to create a comment did not have sufficient permissions.
>
> You can consider shifting to the [pull_request_target](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target) event, where the workflow exists and executes from the base branch of the PR - so it has full access to secrets and write permissions for GITHUB_TOKEN. Please be aware of the warning message in the linked documentation for that triggering event.

Perhaps you want to add respective information to your documentations, @EnricoMi and @PavanMudigondaTR?

**Reply by samus-aran** [reply]
Thank you @ascheman for going the extra mile and posting this publicly, I'm sure this will help others ✨ I wonder if the maintainers would be open to contributions from yourself @ascheman to submit a PR to documentation 🤔 Fabulous work everyone involved 👍
